### PR TITLE
[luci-compute] Introduce initial DepthwiseConv2D

### PR DESCRIPTION
**Pull request by seanshpark**
This will introduce initial compute code for DepthwiseConv2D.

ONE-DCO-1.0-Signed-off-by: SaeHie Park <saehie.park@gmail.com>

### CODE REVIEW

**Comment by seanshpark** [comment]
for #12100 , from  #12099

File: compiler/luci-compute/src/DepthwiseConv2D.cpp [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+  _output_shape.rank(4);
+  _output_shape.dim(0) = input_batches;
+  _output_shape.dim(1) = output_height;
+  _output_shape.dim(2) = output_width;
+  _output_shape.dim(3) = filter_channels_out;
+
+  // TODO compute padding
+  _params.padding_values.height = 0;
+  _params.padding_values.width = 0;
```
Comment (jinevening) [inline compiler/luci-compute/src/DepthwiseConv2D.cpp:70]: Is it ok to hard-code this?

File: compiler/luci-compute/src/DepthwiseConv2D.cpp [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+  if (_params.depth_multiplier != static_cast<int32_t>(filter_channels_out / input_depth))
+    return false; // wrong depth multiplier value
+
+  if (_bias_shape.dim(0).value() != filter_channels_out)
+    return false; // unsupported bias value
+
+  // TODO compute output
+  auto output_height = 0;
+  auto output_width = 0;
```
Comment (jinevening) [inline compiler/luci-compute/src/DepthwiseConv2D.cpp:60]: Are you going to update this code?

File: compiler/luci-compute/src/DepthwiseConv2D.cpp (hunk 1 above)
Comment (seanshpark) [reply compiler/luci-compute/src/DepthwiseConv2D.cpp:70]: There will be actual computation in next PR.
I've left `// TODO compute padding` comment.
If you are curious about the next change, please look at #12099

File: compiler/luci-compute/src/DepthwiseConv2D.cpp (hunk 2 above)
Comment (seanshpark) [reply compiler/luci-compute/src/DepthwiseConv2D.cpp:60]: Yes, `// TODO compute output` comment is for that.

**Comment by seanshpark** [comment]
> This code is very similar to luci-interpreter codes. Is it necessary to re-implement?

The situation: 
- please read  #12100
- luci/pass already has these codes.
- first step is to relocate luci/pass to luci-compute, including this PR
- relocating itself cannot be just copy, as current luci/pass codes are somewhat complicated.
- next step (not sure when) is to refactor luci-interpreter so that to use common computation codes of luci-compute.
- refactoring luci-inpterpreter may not be easy cause of multiple computation backends

Why do we have this situation?
- some of our contributes from the past made this situation with duplicate codes

**Comment by seanshpark** [comment]
> I think if you're going to re-implement new kernels, 

Not re-implementation. I's relocating the luci/pass codes with some cutting the complicated codes in current luci/pass copes with some APIs.

> we need to test them as luci-interpreter does.

That will be in another step. This and upcoming PRs are relocating so that these codes itself is going to be tested from existing luci/pass and other integration tests.

Currently adding also unit tests is too much for me.

**Comment by jinevening** [comment]
Thanks for the replies. When I first see this PR, I thought that code is quite different from the one in the original pass, so I didn't recognize that it's just relocation. I understand it is difficult to extract computation code from the original code. I'll approve the PR as it would have been tested on the draft.

`luci-compute` is a separate module (independent of luci/pass), so I think unittest for each kernel is necessary. I made https://github.com/Samsung/ONE/issues/12150 for that task.

**Review by jinevening — APPROVED** [review]
LGTM

**Review by mhs4670go — APPROVED** [review]
LGTM

File: compiler/luci-compute/include/luci_compute/DepthwiseConv2D.h [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  void output(float *data) { _output_data = data; }
+
+public:
+  bool prepare(void);
+  const loco::TensorShape &outputShape(void) const { return _output_shape; }
+  void compute(void);
+
+private:
+  // param to pass to compute kernel
+  DepthwiseParams _params;
```
Comment (mhs4670go) [inline compiler/luci-compute/include/luci_compute/DepthwiseConv2D.h:64]: @seanshpark Our static analysis tool complains about this line for not initializing `_params` inside the default constructor. And so does `FullyConnected.h`.

Comment (seanshpark) [reply compiler/luci-compute/include/luci_compute/DepthwiseConv2D.h:64]: I'll prepare a change;

**Pull request merged**